README.md

```diff
@@ -1,335 +1,446 @@
+
+~~~
+apt::key { 'puppetlabs':
+  id       => '47B320EB4C7C375AA9DAE1A01054B7A24BD6EC30',
+  server   => 'pgp.mit.edu',
+  options  => 'http-proxy="http://proxyuser:proxypass@example.org:3128"',
+  priority => '50',
+}
+~~~
+
+### Prioritize backports
+
+~~~
+class { 'apt::backports':
+  pin => 500,
+}
+~~~
+
+By default, the `apt::backports` class drops a pin file for backports, pinning it to a priority of 200. This is lower than the normal default of 500, so packages with `ensure => latest` don't get upgraded from backports without your explicit permission.
+
+If you raise the priority through the `pin` parameter to 500, normal policy goes into effect and Apt installs or upgrades to the newest version. This means that if a package is available from backports, it and its dependencies are pulled in from backports unless you explicitly set the `ensure` attribute of the `package` resource to `installed`/`present` or a specific version.
+
+### Update the list of packages
+
+By default, Puppet runs `apt-get update` on the first Puppet run after you include the `apt` class, and anytime `notify  => Exec['apt_update']` occurs; i.e., whenever config files get updated or other relevant changes occur. If you set `update['frequency']` to 'always', the update runs on every Puppet run. You can also set `update['frequency']` to 'daily' or 'weekly':
+
+~~~
+class { 'apt':
+  update => {
+    frequency => 'daily',
+  },
+}
+~~~
+
+### Pin a specific release
+
+~~~
+apt::pin { 'karmic': priority => 700 }
+apt::pin { 'karmic-updates': priority => 700 }
+apt::pin { 'karmic-security': priority => 700 }
+~~~
+
+You can also specify more complex pins using distribution properties:
+
+~~~
+apt::pin { 'stable':
+  priority        => -10,
+  originator      => 'Debian',
+  release_version => '3.0',
+  component       => 'main',
+  label           => 'Debian'
+}
+~~~
+
+To pin multiple packages, pass them to the `packages` parameter as an array or a space-delimited string.
+
+### Add a PPA (Personal Package Archive) repository
+
+~~~
+apt::ppa { 'ppa:drizzle-developers/ppa': }
+~~~
+
+### Add an Apt source to `/etc/apt/sources.list.d/`
+
+~~~
+apt::source { 'debian_unstable':
+  comment  => 'This is the iWeb Debian unstable mirror',
+  location => 'http://debian.mirror.iweb.ca/debian/',
+  release  => 'unstable',
+  repos    => 'main contrib non-free',
+  pin      => '-10',
+  key      => {
+    'id'     => 'A1BD8E9D78F7FE5C3E65D8AF8B48AD6246925553',
+    'server' => 'subkeys.pgp.net',
+  },
+  include  => {
+    'src' => true,
+    'deb' => true,
+  },
+}
+~~~
+
+To use the Puppet Labs Apt repository as a source:
+
+~~~
+apt::source { 'puppetlabs':
+  location => 'http://apt.puppetlabs.com',
+  repos    => 'main',
+  key      => {
+    'id'     => '47B320EB4C7C375AA9DAE1A01054B7A24BD6EC30',
+    'server' => 'pgp.mit.edu',
+  },
+},
+~~~
+
+### Configure Apt from Hiera
+
+~~~yaml
```

Comment:
Docs style at this point is triple tildes for consistency and compatibility throughout our docs. Language-specific highlighting doesn't work on our main docs site.